### PR TITLE
Add a general `read_netcdf` function to read corrupted SOFA data

### PR DESCRIPTION
**Pull request by f-brinkmann**
- [x] refactor `Sofa._update_convention` to `Sofa._reset_convention`. The update functionality is already handled elsewhere
- [x] add `read_netcdf`
- [x] rename `Sofa._protected` to `Sofa.protected` and add explaining docstring
- [x] check where the use of `Sofa._reset_convention` is really required
- [x] Improve error messages when reading Sofa files
- [x] Refer to `Sofa.protected` when attempting to write read only data
- [x] add testing

### CODE REVIEW

**Review by mberz — CHANGES_REQUESTED** [review]
Looks good so far, I just don't like the name of the function ;)

File: sofar/io.py [hunk 1]
```diff
@@ -32,7 +31,7 @@ def read_sofa(filename, verify=True, verbose=True):
     Returns
     -------
     sofa : Sofa
-        The SOFA object filled with the default values of the convention.
+        A SOFA object containing the data from `filename`.
```
[the remark this answers is not in the document]
Comment (f-brinkmann) [reply sofar/io.py:34]: true

File: sofar/io.py [hunk 2]
```diff
@@ -52,8 +51,62 @@ def read_sofa(filename, verify=True, verbose=True):
        will be a scalar inside SOFA objects after reading from disk.
     """
 
+    return _read_netcdf(filename, verify, verbose, mode="sofa")
+
+
+def read_netcdf(filename):
```
[the remark this answers is not in the document]
Comment (f-brinkmann) [reply sofar/io.py:57]: changed it as suggested

File: sofar/io.py [hunk 3]
```diff
@@ -52,8 +51,62 @@ def read_sofa(filename, verify=True, verbose=True):
...
+
+    Parameters
+    ----------
+    filename : str
+        The full path to the NetCDF data.
+
+    Returns
+    -------
+    sofa : Sofa
+        A SOFA object containing the data from `filename`.
```
[the remark this answers is not in the document]
Comment (f-brinkmann) [reply sofar/io.py:84]: Yes

File: sofar/io.py [hunk 4]
```diff
@@ -52,8 +51,62 @@ def read_sofa(filename, verify=True, verbose=True):
        will be a scalar inside SOFA objects after reading from disk.
     """
 
+    return _read_netcdf(filename, verify, verbose, mode="sofa")
+
+
+def read_netcdf(filename):
+    """
+    Read NetCDF file from disk and convert it to SOFA object.
```
[the remark this answers is not in the document]
Comment (f-brinkmann) [reply sofar/io.py:59]: How about this?

**Review by mberz — APPROVED** [review]

**Pull request merged**